src/traffic/core/flight.py

```diff
@@ -1879,6 +1879,203 @@ def cascaded_filters(
+    ) -> "Flight":
+        """Filter a parameter with a clustering approach. Datapoints are clustered based on
+        difference in time and parameter value. If the cluster is larger than the defined groupsize
+        the datapoints are kept, otherwise they are removed.
+
+        :param paracol: Name of the column to filter.
+        :param groupsize: Minimum size of a cluster in order not to be removed
+        :param timediff_big: Threshold above which a difference in time is considered a new cluster.
+        :param paradiff_big: Theshold above which a difference in parameter value is considered a
+                             new cluster.
+        :param timecol: Name of the column containing the timestamp information,by default
+                        "timestamp".
+        """
+
+        data = self.data.reset_index(drop=True)
+        data = data.drop_duplicates([timecol], keep="last")
+        if timediffs_big is None:
+            timediffs_big = [60] * len(paracols)
+
+        for paracol, groupsize, timediff_big, paradiff_big in zip(
+            paracols, groupsizes, timediffs_big, paradiffs_big
+        ):
+            if paracol == "onground":
+                statechange = data[paracol].diff().astype(bool)
+                data["group"] = statechange.eq(True).cumsum()
+                groups = data["group"].value_counts()
+                keepers = groups[groups > groupsize].index.tolist()
+                data[paracol] = data[paracol].where(
+                    data["group"].isin(keepers), float("NaN")
+                )
+
+            else:
+                timediff = data[timecol].diff().astype("timedelta64[s]")
+                temp_index = data.index
+                temp_values = data[paracol].dropna()
+                paradiff = abs(temp_values.diff().reindex(temp_index))
+                bigdiff = pd.Series(
+                    np.where(
+                        ((timediff > timediff_big) | (paradiff > paradiff_big)),
+                        True,
+                        False,
+                    )
+                )
+                data["group"] = bigdiff.eq(True).cumsum()
+                groups = data[data[paracol].notna()]["group"].value_counts()
+                keepers = groups[groups > groupsize].index.tolist()
+                data[paracol] = data[paracol].where(
+                    data["group"].isin(keepers), float("NaN")
+                )
+
+            data = data.drop(columns=["group"])
+        return self.__class__(data)
+
+    def smoothing(
+        self, paracols: List[str], kernel_sizes: List[int]
+    ) -> "Flight":
+        """Smoothen a parameter using a rolling mean with a specified kernel size.
+
+        :param paracol: Name of the column to smooth.
+        :param kernel_size: The size of the kernel to use for the rolling mean.
+        """
+        for paracol, kernel_size in zip(paracols, kernel_sizes):
+            data = self.data.reset_index(drop=True)
```

Comment:
Shouldn't this line be outside the loop?
Also, why not name this one filter_mean? (the only difference with filter_median is the mean method, right?)